packages/react/src/components/table/DataTable.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useEffect, useImperativeHandle, useMemo, useState } from 'react';
+
+// import core base styles
+import 'hds-core';
+import styles from './Table.module.scss';
+import { Table } from './Table';
+import { Checkbox } from '../checkbox';
+import { useTheme } from '../../hooks/useTheme';
+
+type Header = {
+  key: string;
+  headerName: string;
+  sortIconType?: 'string' | 'other';
+  transform?: ({ args }: any) => string | JSX.Element; // eslint-disable-line @typescript-eslint/no-explicit-any
+};
+
+export interface TableCustomTheme {
+  '--background-color'?: string;
+}
+
+export type DataTableProps = React.ComponentPropsWithoutRef<'table'> & {
+  cellConfig: {
+    cols: Array<Header>;
+    sortingEnabled?: boolean;
+    initialSortingColumnKey?: string; // undefined -> neutral order for all columns
+    initialSortingOrder?: 'asc' | 'desc';
+    ariaLabelSortButtonNeutral?: string;
+    ariaLabelSortButtonAscending?: string;
+    ariaLabelSortButtonDescending?: string;
+    indexKey: string; // column key used as unique identifier for row
+    renderIndexCol?: boolean; // whether index colum is rendered in table. Defaults to true.
+  };
+  rows: Array<object>;
+  verticalHeaders?: Array<Header>;
+  variant?: 'dark' | 'light';
+  dense?: boolean;
+  zebra?: boolean;
+  caption?: string | React.ReactNode;
+  verticalLines?: boolean;
+  textAlignContentRight?: boolean; // defaults to false -> text is aligned left
+  checkboxSelection?: boolean;
+  ariaLabelCheckboxSelection?: string;
+  setSelections?: Function; // Callback that gets called with all selected row id values
+  initiallySelectedRows?: any[]; // Initially selected rows. Apply corresponding indexKey values here.
```

Comment:
Could the row type be an object with arbitrary key and with known value types: string, number, date, etc.?

Reply:
The selectedRow type is actually different from the basic row type. This is only a list of index key values, like:
`[1000, 1001, 1002]`. Maybe we could restrict it string and number?

Reply:
- [ ] Create and use SelectedRow typescript type